tests/test_simulator_device.py

```diff
@@ -529,3 +605,58 @@ def circuit():
         # With 3 samples we are guaranteed to see a difference between
         # an estimated variance an an analytically calculated one
         assert var != 1.0
+
+@pytest.mark.parametrize("shots,analytic", [(100, True)])
+class TestSample:
+    """Test sampling."""
+
+    def test_probability_sum_error(self, simulator_device_1_wire):
+        """Test that an error is raised if the probabilites in the 
+        internal state do not sum up to 1."""
+
+        simulator_device_1_wire.state = np.array([1, 1])
+        with pytest.raises(ValueError, match="Probabilites in sampling must sum up to 1."):
+            simulator_device_1_wire.sample("PauliZ", wires=[0], par=[])
```

Comment:
Would this be caused by the Cirq simulator itself? :thinking:

Reply:
As in, this is never something that could be triggered by the user? (is it worth catching, in that case?)

Reply:
If you look at the lines below, then you see that I have to normalize the probability vector for numpy's sampling to work. This is caused by small deviations from unity in the Cirq simulator (and probably my postprocessing). As this could possibly hide errors I decided to check that it's not too far off and to throw an error if that was the case.

Do you think this could come from the conversion to a numpy array with `float` dtype? And do you think it is better not to check for an error here?

Reply:
Okay, after thinking about it I think you're right. We should trust the Cirq simulator to not give us crap.